edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/EdgeAgentDockerModule.cs

```diff
@@ -36,6 +53,8 @@ public override int GetHashCode()
                 hashCode = (hashCode * 397) ^ (this.Version != null ? this.Version.GetHashCode() : 0);
                 hashCode = (hashCode * 397) ^ (this.Type != null ? this.Type.GetHashCode() : 0);
                 hashCode = (hashCode * 397) ^ (this.Config?.Image != null ? this.Config.Image.GetHashCode() : 0);
+                hashCode = (hashCode * 397) ^ (this.Config?.CreateOptions != null ? this.Config.CreateOptions.GetHashCode() : 0);
+                hashCode = (hashCode * 397) ^ (this.Env != null ? this.Env.GetHashCode() : 0);
```

Comment:
Content in dictionaries can be the same, but their hash code can be different.  I think we don't need to consider this.Env in GetHashCode.

Reply:
Yeah I wasn't quite sure of the best thing to do here. You think we don't need `Env`, but we _do_ need `CreateOptions`?

Reply:
We can't use this.Config.CreateOptions.GetHashCode() as it doesn't override GetHashCode() method and use default object GetHashCode method, which means the content of 2 CreateOptions can be same but their GetHashCode will return different values.  Same case as Dictionary.

Reply:
Ah right...I'm mixing reference- and value-type hashing semantics here. I'll remove...

Reply:
Removed